### PR TITLE
another drop — for review

### PR DESCRIPTION
**Pull request by costa**

### CODE REVIEW

File: lib/ffmprb.rb [hunk 1]
```diff
@@ -1,12 +1,12 @@
+require 'ffmprb/execution'
 require 'ffmprb/file'
 require 'ffmprb/filter'
+require 'ffmprb/find_silence'
 require 'ffmprb/process'
 require 'ffmprb/util'
 require 'ffmprb/version'
```
Comment (tanin) [inline lib/ffmprb.rb:7]: require recursively, something like:

`Dir["#{File.dirname(__FILE__)}/ffmprb/**/*.rb"].each { |f| require f }`

File: lib/ffmprb.rb [hunk 2]
```diff
@@ -19,20 +19,33 @@ module Ffmprb
...
   end
 
+  Util.ffmpeg_cmd = ['ffmpeg']
+  Util.ffprobe_cmd = ['ffprobe']
+
+  Process.duck_audio_hi = 0.9
+  Process.duck_audio_lo = 0.1
+  Process.duck_audio_transition_sec = 1
+  Process.duck_audio_silent_min_sec = 3
+  Filter.silence_noise_max_db = -40
```
Comment (tanin) [inline lib/ffmprb.rb:29]: comments

File: lib/ffmprb.rb [hunk 3]
```diff
@@ -19,20 +19,33 @@ module Ffmprb
...
+
   Util::IoBuffer.blocks_max = 1024
   Util::IoBuffer.block_size = 64*1024
   Util::IoBuffer.timeout = 9
 
   class << self
 
+    # NOTE the form with the block returns the result of #run
+    # NOTE the form without the block returns the process (before it is run)
     def process(*args, &blk)
```
Comment (tanin) [inline lib/ffmprb.rb:39]: How about 2 different methods process and process! ? The first returns process instance the second the result of #run

File: lib/ffmprb/file.rb [hunk 4]
```diff
@@ -117,24 +139,25 @@ def resolution
...
       video = File.temp('.jpg')  if video == true
 
       Ffmprb.logger.debug "Snap shooting files, video path: #{video ? video.path : 'NONE'}, audio path: #{audio ? audio.path : 'NONE'}"
 
-      raise Error, "Incorrect output extname (must be .jpg)"  unless !video || video.extname =~ /jpe?g$/
-      raise Error, "Incorrect audio extname (must be .mp3)"  unless !audio || audio.extname =~ /mp3$/
-      raise Error, "Can sample either video OR audio UNLESS a block is given"  unless block_given? || (!!audio != !!video)
+      raise Error, "Incorrect output extname (must be .jpg)"  unless !video || video.channel?(:video) && !video.channel?(:audio)
+      raise Error, "Incorrect audio extname (must be .mp3)"  unless !audio || audio.channel?(:audio) && !audio.channel?(:video)
+      raise Error, "Can sample either video OR audio UNLESS a block is given"  unless block_given? || !!audio != !!video
```
Comment (tanin) [inline lib/ffmprb/file.rb:155]: Maybe add methods for the unless conditions?

File: lib/ffmprb/process.rb [hunk 5]
```diff
@@ -5,43 +5,50 @@ module Ffmprb
...
-            next  if silent.end_at && silent.start_at && (silent.end_at - silent.start_at) < 3
-            ducked_overlay_volume.merge!(
-              (silent.start_at - 0.5) => 0.1,
-              (silent.start_at + 0.5) => 0.9
-            )  if silent.start_at
-            ducked_overlay_volume.merge!(
-              (silent.end_at - 0.5) => 0.9,
-              (silent.end_at + 0.5) => 0.1
-            )  if silent.end_at
+      def duck_audio(av_main_i, a_overlay_i, silence, av_main_o, video: {resolution: Ffmprb::QVGA, fps: 30})
```
Comment (tanin) [inline lib/ffmprb/process.rb:25]: A bit strange that duck_audio is caring about the video resolution, are you sure it should be here?

**Comment by costa** [comment]
@tanin I'm responding to this code review in https://github.com/showbox-oss/ffmprb/pull/4

File: lib/ffmprb.rb (hunk 1 above)
Comment (costa) [inline lib/ffmprb.rb:7]: fixed

File: lib/ffmprb.rb (hunk 2 above)
Comment (costa) [inline lib/ffmprb.rb:29]: better naming coming instead

File: lib/ffmprb.rb (hunk 3 above)
Comment (costa) [inline lib/ffmprb.rb:39]: added an XXX comment, not sure the second form is needed at all

File: lib/ffmprb/file.rb (hunk 4 above)
Comment (costa) [inline lib/ffmprb/file.rb:155]: umm, no, too marginal a case, moved the whole method to another file though

File: lib/ffmprb/process.rb (hunk 5 above)
Comment (costa) [inline lib/ffmprb/process.rb:25]: you're right, it is strange, refactoring is on its way

**Pull request closed without merging**